pkg/build/buildkit/errors.go

```diff
@@ -49,54 +47,83 @@ func (e CommandErr) Error() string {
 }
 
-// GetErrorMessage returns the parsed error message
-func GetErrorMessage(err error, tag string) error {
+var (
+	oktetoRemoteCLIImage = "okteto/okteto"
+	dockerhubRegistry    = "docker.io"
+)
+
+func isOktetoRemoteImage(image string) bool {
+	return strings.Contains(image, oktetoRemoteCLIImage) && strings.Contains(image, dockerhubRegistry)
+}
+
+func isOktetoRemoteForkImage(image string) bool {
+	return strings.Contains(image, oktetoRemoteCLIImage)
+}
+
+func isImageIsNotAccessibleErr(err error) bool {
+	return isLoggedIntoRegistryButDontHavePermissions(err) ||
+		isNotLoggedIntoRegistry(err) ||
+		isPullAccessDenied(err) ||
+		isNotFound(err) ||
+		isHostNotFound(err)
+}
+
+// GetSolveErrorMessage returns the parsed error message
+func GetSolveErrorMessage(err error) error {
 	if err == nil {
 		return nil
 	}
 
-	imageCtrl := registry.NewImageCtrl(okteto.Config{})
-	imageRegistry, imageTag := imageCtrl.GetRegistryAndRepo(tag)
+	imageFromError := extractImageFromError(err)
 	switch {
-	case isLoggedIntoRegistryButDontHavePermissions(err):
-		err = oktetoErrors.UserError{
-			E:    fmt.Errorf("failed to push image '%s': You are not authorized to push image '%s'", tag, imageTag),
-			Hint: fmt.Sprintf("Please log in into the registry '%s' with a user with push permissions to '%s' or use another image.", imageRegistry, imageTag),
-		}
-	case isNotLoggedIntoRegistry(err):
-		err = oktetoErrors.UserError{
-			E:    fmt.Errorf("failed to push image '%s': You are not authorized to push image '%s'", tag, imageTag),
-			Hint: fmt.Sprintf("Log in into the registry '%s' and verify that you have permissions to push the image '%s'.", imageRegistry, imageTag),
-		}
 	case isBuildkitServiceUnavailable(err):
 		err = oktetoErrors.UserError{
 			E:    fmt.Errorf("buildkit service is not available at the moment"),
 			Hint: "Please try again later.",
 		}
-	case isPullAccessDenied(err):
-		if imageTag == "" {
-			imageTag = extractImageTagFromPullAccessDeniedError(err)
-		}
+	case isImageIsNotAccessibleErr(err):
 		err = oktetoErrors.UserError{
-			E:    fmt.Errorf("failed to pull image '%s'. The repository is not accessible or it does not exist", imageTag),
-			Hint: fmt.Sprintf("Please verify the name of the image '%s' to make sure it exists.", imageTag),
+			E: fmt.Errorf("the image '%s' is not accessible or it does not exist", imageFromError),
+			Hint: `Please verify the name of the image to make sure it exists.
+    When using private registries, make sure Okteto Registry Credentials are correctly configured.
+    See more at: https://www.okteto.com/docs/admin/registry-credentials/`,
+		}
+
+		if isOktetoRemoteImage(imageFromError) {
+			err = oktetoErrors.UserError{
+				E: fmt.Errorf("the image '%s' is not accessible or it does not exist", imageFromError),
+				Hint: `Please verify you have access to dockerhub.
+    If you are using airgapped environment check our docs on how to use Okteto Remote in airgapped environments: [TBD]`,
```

Comment:
probably link to docs of airgapped woulld be https://github.com/okteto/docs/pull/885, let me know what link should we use for now instead? @ifbyol @pchico83

Reply:
Link will be available at https://www.okteto.com/docs/self-hosted/manage/air-gapped/